src/gui/osutils/macutils/MacUtils.cpp

```diff
@@ -152,11 +152,18 @@ bool MacUtils::isCapslockEnabled()
 
 void MacUtils::setUserInputProtection(bool enable)
 {
+    static bool secureInputEnabled = false;
     if (enable) {
+        // Always keep the internal counter set to 1
```

Comment:
This is kind of arcane. There should be a more thorough explanation in the comment as to what this does and why it's needed.

Reply:
The api with macos is so bad. It keeps an internal counter that you have to decrement to zero to disable the feature. There is no way to determine the counter value. There is also no way to determine if your app is still blocking.

Reply:
Put that in the comment and we're good. 😉

Reply:
Thank you for the fix @droidmonkey! Would love to see it in one of the next versions. Here's my take for a more comprehensive comment, hope to save you some work.

```suggestion
        /*
         * MacOS keeps a single counter over all apps that needs to be zero to disable secure input. By never going
         * higher than 1 internally this makes sure secure input doesn't stay active after calling this function
         * multiple times.
         */
```

Reply:
Thank you!